### ISSUE TITLE
Ability to add custom properties or free text input to a batch change/changeset

### ISSUE THREAD

**Issue by courier-new** [sourcegraph/sourcegraph#43400]
It would be useful if users were able to add custom notes or configure custom properties for batch changes or changesets, like a team name, slack channel, link, status note, etc. These fields could be surfaced from list views and enable custom filtering/sorting, as well. Lots of customers do this already today in their own external spreadsheets and tools. Bringing this into Sourcegraph would save on a lot of manual labor and reduce the need for duplication.

**Comment by chrispine** [comment]
Is this a dupe of #43396 ?

**Comment by courier-new** [comment]
No, but we talked about them both in the same conversation at our offsite and technically speaking this one might be the generic superset of that one. This one is about freeform notes of potentially any data type, on any batch changes "entity", and that one's specifically scoped to assigning tags to changesets.
